mobile/src/reducers/calendar/calendar-intervals-builder.spec.ts

```diff
@@ -177,27 +204,153 @@ describe('CalendarIntervalsBuilder', () => {
         expect(intervals.length).toBe(1);
         expect(intervals[0].intervalType).toBe('startInterval');
         expect(intervals[0].eventType).toBe(CalendarEventsType.Vacation);
-        
+        expect(intervals[0].startDate).toBe(event1.dates.startDate);
+        expect(intervals[0].endDate).toBe(event1.dates.endDate);
+
         intervals = intervalsModel.get(moment({ day: 2, month: 0, year: 2018 }));
 
         expect(intervals.length).toBe(2);
         expect(intervals[0].intervalType).toBe('interval');
         expect(intervals[0].eventType).toBe(CalendarEventsType.Vacation);
+        expect(intervals[0].startDate).toBe(event1.dates.startDate);
+        expect(intervals[0].endDate).toBe(event1.dates.endDate);
         expect(intervals[1].intervalType).toBe('startInterval');
         expect(intervals[1].eventType).toBe(CalendarEventsType.SickLeave);
+        expect(intervals[1].startDate).toBe(event2.dates.startDate);
+        expect(intervals[1].endDate).toBe(event2.dates.endDate);
 
         intervals = intervalsModel.get(moment({ day: 3, month: 0, year: 2018 }));
 
         expect(intervals.length).toBe(2);
         expect(intervals[0].intervalType).toBe('endInterval');
         expect(intervals[0].eventType).toBe(CalendarEventsType.Vacation);
+        expect(intervals[0].startDate).toBe(event1.dates.startDate);
+        expect(intervals[0].endDate).toBe(event1.dates.endDate);
         expect(intervals[1].intervalType).toBe('interval');
         expect(intervals[1].eventType).toBe(CalendarEventsType.SickLeave);
+        expect(intervals[1].startDate).toBe(event2.dates.startDate);
+        expect(intervals[1].endDate).toBe(event2.dates.endDate);
 
         intervals = intervalsModel.get(moment({ day: 4, month: 0, year: 2018 }));
 
         expect(intervals.length).toBe(1);
         expect(intervals[0].intervalType).toBe('endInterval');
         expect(intervals[0].eventType).toBe(CalendarEventsType.SickLeave);
+        expect(intervals[0].startDate).toBe(event2.dates.startDate);
+        expect(intervals[0].endDate).toBe(event2.dates.endDate);
+    });
+
+    describe('dayoff', () => {
+
+        const testDayoff = (testedEventType: CalendarEventsType.Dayoff | CalendarEventsType.AdditionalWork) => {
+            describe(testedEventType, () => {
+                it('should return interval left boundary, if startWorkingHour between (0, 4] and finishWorkingHour is 0', () => {
```

Comment:
finishWorkingHour should not be less then startWorkingHour. Does it actually happen?

Reply:
As I realized, if finishWorkingHour is between 0 and 4, with finishWorkingHour = 0, it means a first half?